scraper/src/github-scraper/utils.ts

```diff
@@ -0,0 +1,142 @@
+};
+
+export async function calculateTurnaroundTime(event: PullRequestEvent) {
+  const user: string = event.payload.pull_request.user.login;
+  const mergedAt: Date = parseISODate(event.payload.pull_request.merged_at);
+  const createdAt: Date = parseISODate(event.payload.pull_request.created_at);
+
+  const linkedIssues: [string, string][] = [];
+  const body = event.payload.pull_request.body || "";
+  const regex =
+    /(fix|fixes|fixed|close|closes|closed|resolve|resolves|resolved) ([\w\/.-]*)(#\d+)/gi;
+  let match: RegExpExecArray | null;
+
+  while ((match = regex.exec(body)) !== null) {
+    linkedIssues.push([match[2], match[3]]);
+  }
+
+  const prTimelineResponse = await octokit.request(
+    `GET ${event.payload?.pull_request?.issue_url}/timeline`,
+  );
+
+  const prTimeline = prTimelineResponse.data;
+
+  prTimeline.forEach((action: Action) => {
+    if (
+      action.event === "cross-referenced" &&
+      action.source.type === "issue" &&
+      !action.source.issue.pull_request
+    ) {
+      linkedIssues.push([
+        action.source.issue.repository.full_name,
+        `#${action.source.issue.number}`,
+      ]);
+    }
+
+    if (action.event === "connected") {
+      // TODO: currently there is no way to get the issue number from the timeline, handle this case while moving to graphql
+    }
+  });
+  const uniqueLinkedIssues: [string, string][] = Array.from(
+    new Set(linkedIssues.map((issue) => JSON.stringify(issue))),
+  ).map((item) => JSON.parse(item) as [string, string]);
+  const assignedAts: { issue: string; time: Date }[] = [];
+
+  for (const [org_repo, issue] of uniqueLinkedIssues) {
+    const org = org_repo.split("/")[0] || event.repo.name.split("/")[0];
+    const repo = org_repo.split("/")[-1] || event.repo.name.split("/")[1];
+    const issueNumber = parseInt(issue.split("#")[1]);
+
+    const issueTimelineResponse = await octokit.request(
+      "GET /repos/{owner}/{repo}/issues/{issue_number}/timeline",
+      {
+        owner: org,
+        repo: repo,
+        issue_number: issueNumber,
+      },
+    );
+
+    const issueTimeline = issueTimelineResponse.data;
+    issueTimeline.forEach((action: Action) => {
+      if (action.event === "assigned" && action.assignee.login === user) {
+        assignedAts.push({
+          issue: `${org}/${repo}#${issueNumber}`,
+          time: parseISODate(action.created_at),
+        });
+      }
+
+      if (action.event === "unassigned" && action.assignee.login === user) {
+        assignedAts.pop();
+      }
+    });
+  }
+
+  const assignedAt: Date | null =
+    assignedAts.length === 0
+      ? null
+      : assignedAts.reduce((min, current) =>
+          current.time < min.time ? current : min,
+        ).time;
+  const turnaroundTime =
+    (mergedAt.getTime() - (assignedAt || createdAt.getTime()).valueOf()) / 1000;
+  return turnaroundTime;
+}
+
+export async function resolve_autonomy_responsibility(
```

Comment:
Use proper naming conventins. `snake_case` is used in python. 

```suggestion
export async function resolveAutonomyResponsibility(
```